packages/core/src/browser/context-key-service.ts

```diff
@@ -54,4 +59,11 @@ export class ContextKeyService {
         return true;
     }
 
+    /**
+     * It should be implemented by an extension, e.g. by the monaco extension.
+     */
+    deserialize(when: string): ContextKeyExpr {
```

Comment:
How about `Set<string>` as a return type?

Reply:
let's rename it to something else, like `parseKeys(when: string): Set<string>`

`ContextKeyExpr` does not look like complete api, and i don't think we want it to core right now

Reply:
done